### PR TITLE
IOGX Integration

### PR DESCRIPTION
**Pull request by zeme-wana**

### CODE REVIEW

File: nix/haskell-project.nix [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+let
+  lib = pkgs.lib;
+  isDarwin = pkgs.stdenv.hostPlatform.isDarwin;
+  isCross = pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform;
+
+  antaeus-module = { config, ... }: {
+    packages = {
+      # Things that need plutus-tx-plugin
+      freer-extras.package.buildable = !isCross;
+      cardano-node-emulator.package.buildable = !isCross;
```
Comment (james-iohk) [inline nix/haskell-project.nix:19]: may not need:
- `cardano-node-emulator`
- `cardano-streaming`
- `plutus-pab-executables`

File: flake.nix [hunk 2]
```diff
@@ -1,153 +1,35 @@
...
-      {
-        # Here we say that we want the "devcontainer" cell block of the antaeus cell
-        # (which contains a number of buildable derivations) to be exposed
-        # by the flake and accessible via nix build (or nix run).
-        packages = inputs.std.harvest inputs.self [ "antaeus" "devcontainer" ];
-      }
-      {
-        hydraJobs = inputs.std.harvest inputs.self [ "automation" "hydra-jobs" ];
-      };
+    inputs.iogx.mkFlake {
```
Comment (shlevy) [inline flake.nix:9]: Do we need all the defaults set?

File: nix/haskell-project.nix [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  sha256map = {
+    "https://github.com/input-output-hk/cardano-node"."a158a679690ed8b003ee06e1216ac8acd5ab823d" = "sha256-uY7wPyCgKuIZcGu0+vGacjGw2kox8H5ZsVGsfTNtU0c=";
+    "https://github.com/james-iohk/cardano-node"."adf50dc5de3d44bdb5c3dc0b28e18b3a5477f36c" = "18yhmfa95sfry9jsgv9rg1giv73235wwjvw7qr3jximj88gprakn";
+  };
+
+  # Configuration settings needed for cabal configure to work when cross compiling
+  # for windows. We can't use `modules` for these as `modules` are only applied
+  # after cabal has been configured.
+  cabalProjectLocal = lib.optionalString pkgs.stdenv.hostPlatform.isWindows ''
```
Comment (shlevy) [inline nix/haskell-project.nix:85]: I wonder if there's a way we can fix this upstream in h.nix, generating cabal.project boilerplate seems wrong

File: nix/shell-module.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  # the shell:
+  # packages = [
+  #   haskell-nix-project.hsPkgs.cardano-cli.components.exes.cardano-cli
+  #   haskell-nix-project.hsPkgs.cardano-node.components.exes.cardano-node
+  # ];
+  # Be careful not to refence the project's own haskell packages.
+, haskell-nix-project
+}:
+
+{
```
Comment (shlevy) [inline nix/shell-module.nix:36]: Why aren't these just added into the shell?

**Review by shlevy — COMMENTED** [review]
Comments only on IOGX/antaeus interaction, not on core IOGX interface itself.

File: nix/shell-module.nix (hunk 4 above)
Comment (zeme-wana) [reply nix/shell-module.nix:36]: You mean the executables? 
I suppose that can come in handy, I will add them!
(We need the envvars too though, other components expect them to be set)

**Review by james-iohk — APPROVED** [review]
LGTM, thanks!

**Pull request merged**